### PR TITLE
🌱 Add `ClusterExtension` package uniqueness test

### PR DESCRIPTION
**Pull request by m1kola**
# Description

Follow up to #774

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: test/e2e/cluster_extension_admission_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+)
+
+func TestClusterExtensionPackageUniqueness(t *testing.T) {
```
Comment (m1kola) [inline test/e2e/cluster_extension_admission_test.go:16]: Feels a bit like we are testing upstream Kubernetes... Is this test worth having?

Comment (m1kola) [reply test/e2e/cluster_extension_admission_test.go:16]: On the other hand - I got CEL expression wrong at first while working on #774: I blocked edits :) Given that I think it makes sense.

**Review by tmshort — COMMENTED** [review]
/lgtm

**Review by tmshort — APPROVED** [review]

**Comment by m1kola** [comment]
I think I introduced a flake here. [Example](https://github.com/operator-framework/operator-controller/actions/runs/8844983275/job/24287959651).

I'll look into it.

**Pull request merged**